pkg/co2/source/ukgrid.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type UKGrid struct {
+	collectionSupported bool
+}
+
+// Define a struct to hold the json data
+type Intensity struct {
+	Data []struct {
+		From      string `json:"from"`
+		To        string `json:"to"`
+		Intensity struct {
+			Forecast int    `json:"forecast"`
+			Actual   int    `json:"actual"`
+			Index    string `json:"index"`
+		} `json:"intensity"`
+	} `json:"data"`
+}
+
+func (d *UKGrid) Init() error {
+	d.collectionSupported = false
+	return nil
+}
+
+func (d *UKGrid) Shutdown() bool {
+	return true
+}
+
+func (d *UKGrid) GetCarbonIntensity(url string) (int64, error) {
```

Comment:
@jbu can you also have a result cache here so the HTTP request happens less frequently?

Reply:
Definitely. As I wanted to spark discussion first I haven't done it. I'll add it when I get a chance.